packages/headless/src/features/search-parameters/search-parameter-serializer.ts

```diff
@@ -20,7 +23,7 @@ function serialize(obj: SearchParameters) {
     .join(delimiter);
 }
 
-function serializePair(pair: [string, unknown]) {
+function serializePair(pair: [string, string | number | boolean]) {
```

Comment:
It's the expected type for `encodeURIComponent`.

It could however introduce bugs for the above line in that function, since those can be object 🤔 
I'll move the cast to only the specific line that needs it, because at that point in the function, it has to be one of those type.